tools/db_bench_tool.cc

```diff
@@ -6246,15 +6445,24 @@ class Benchmark {
 
     std::unique_ptr<const char[]> key_guard;
     Slice key = AllocateKey(&key_guard);
+    std::unique_ptr<char[]> ts_guard;
+    if (user_timestamp_size_ > 0) {
+      ts_guard.reset(new char[user_timestamp_size_]);
+    }
     // the number of iterations is the larger of read_ or write_
     while (!duration.Done(1)) {
       DB* db = SelectDB(thread);
       GenerateKeyFromInt(thread->rand.Next() % FLAGS_num, FLAGS_num, &key);
+      Slice ts;
+      if (user_timestamp_size_ > 0) {
+        ts = mock_app_clock_->Allocate(ts_guard.get());
```

Comment:
Just to confirm: we're intentionally reading the latest value unconditionally here and in the test cases below, right?

Reply:
Yes. This is due to the operation being read-modify-write. Therefore, we read the latest using the newest timestamp.

Reply:
Yeah that's what I thought :) Thanks!